testproject/Assets/Tests/Manual/HybridScripts/RpcQueueManualTests.cs

```diff
@@ -578,79 +703,92 @@ private void OnSendGlobalCounterClientRpc(int counter)
         }
 
+        private string m_ServerUpdateInfo;
+        private int m_ServerDirectTotalRpcCount;
         /// <summary>
         /// Updates the server text info and host progress bar
         /// </summary>
         private void UpdateServerInfo()
         {
-            string updatedCounters = string.Empty;
+            m_ServerUpdateInfo = string.Empty;
             foreach (var entry in m_ClientSpecificCounters)
             {
-                if (entry.Key == 0 && IsHost)
+                if (entry.Key == NetworkManager.LocalClientId && IsHost)
                 {
-                    updatedCounters += $"Client-ID [{entry.Key}]  Client to Server Rpcs Received: {entry.Value}  |  Broadcast Rpcs Sent:{m_GlobalCounter} -- Direct Rpcs Sent:{m_GlobalDirectCounter}\n";
-                    updatedCounters += $"{nameof(m_MultiParameterValuesCount)} : {m_MultiParameterValuesCount}  |  {nameof(m_MultiParameterNoneCount)} : {m_MultiParameterNoneCount}\n";
-                    updatedCounters += $"{nameof(m_RpcPerSecond)} : {m_RpcPerSecond}\n ";
+                    m_ServerUpdateInfo += $"Client-ID [{entry.Key}]  Client to Server Rpcs Received: {entry.Value}  |  Broadcast Rpcs Sent:{m_GlobalCounter} -- Direct Rpcs Sent:{(UnitTesting ? m_ServerDirectTotalRpcCount : m_GlobalDirectCounter)}\n";
+                    m_ServerUpdateInfo += $"{nameof(m_MultiParameterValuesCount)} : {m_MultiParameterValuesCount}  |  {nameof(m_MultiParameterNoneCount)} : {m_MultiParameterNoneCount}\n";
+                    m_ServerUpdateInfo += $"{nameof(m_RpcPerSecond)} : {m_RpcPerSecond}\n ";
                 }
                 else
                 {
-                    updatedCounters += $"Client-ID [{entry.Key}]  Client to Server Rpcs Received: {entry.Value}\n";
+                    m_ServerUpdateInfo += $"Client-ID [{entry.Key}]  Client to Server Rpcs Received: {entry.Value}\n";
                 }
             }
 
-            updatedCounters += $"{nameof(m_ClientRpcDirectTestingMode)} : {m_ClientRpcDirectTestingMode}";
+            m_ServerUpdateInfo += $"{nameof(m_ClientRpcDirectTestingMode)} : {m_ClientRpcDirectTestingMode}";
 
-            if (IsHost)
+            if (!UnitTesting)
             {
-                if (m_GlobalDirectCurrentClientIdIndex < m_ClientIds.Count)
+                if (IsHost)
                 {
-                    if (m_ClientProgressBar && m_ClientIndices.Contains(NetworkManager.Singleton.LocalClientId))
+                    if (m_GlobalDirectCurrentClientIdIndex < m_ClientIds.Count)
                     {
-                        m_ClientProgressBar.fillAmount = Mathf.Clamp((2.0f * (float)m_GlobalDirectCounter) * k_ProgressBarDivisor, 0.01f, 1.0f);
+                        if (m_ClientProgressBar && m_ClientIndices.Contains(NetworkManager.LocalClientId))
+                        {
+                            m_ClientProgressBar.fillAmount = Mathf.Clamp((2.0f * (float)m_GlobalDirectCounter) * k_ProgressBarDivisor, 0.01f, 1.0f);
```

Comment:
an innocent operation like this takes a long way in branches to be executed :)
I feel like there could be a logic compression somehow but maybe all those checks are really necessary.

Reply:
**_Please take note of the !UnitTesting on line 778_**

Reply:
This was a manual test that is now a hybrid:
-----------------------------------------------
- It can operate in manual mode
- It can operate in unit test mode
-----------------------------------------------

It will be fine.  :)

Reply:
why go hybrid?

Reply:
It saves fuel of course!

Reply:
in the scope of this PR, I'd appreciate that but this reminds me that we need to have a "dependencies and isolation" conversation soon, internally.

Reply:
In all reality, it was made hybrid so we can perform a visual acceptance test on the acceptance tests.  This is why the end of the unit test version displays the count totals for everything, so one can verify that indeed the counters are all being incremented in the same fashion running as a unit test as it does when ran "manually".

Reply:
if the test was to be fully-automated, it wouldn't need manual testing or visual testing in the first place :)

Reply:
Well perhaps the task was not defined properly... it was to convert the manual tests over to unit tests.
(that leaves a lot of room for interpretation)